python-avd/pyavd/_eos_designs/structured_config/network_services/application_traffic_recognition.py

```diff
@@ -181,74 +167,62 @@ def _filtered_application_classification(self: AvdStructuredConfigNetworkService
                         )
                         raise AristaAvdInvalidInputsError(msg)
 
-                    append_dict_to_list_of_dicts(
-                        list_of_dicts=application_profiles, new_dict=self.inputs.application_classification.application_profiles[application_profile]._as_dict()
-                    )
+                    application_profile_item = self.inputs.application_classification.application_profiles[application_profile]
 
-        output = {"application_profiles": application_profiles}
-        # Now handle categories, applicaations
-        categories = []
-        applications = []
+                    self.structured_config.application_traffic_recognition.application_profiles.append(application_profile_item)
 
-        for application_profile in application_profiles:
-            for category in get(application_profile, "categories", default=[]):
-                if category["name"] not in self.inputs.application_classification.categories:
+        for application_profile in self.structured_config.application_traffic_recognition.application_profiles:
+            for category in application_profile.categories:
+                if category.name and category.name not in self.inputs.application_classification.categories:
                     msg = (
-                        f"The application profile {application_profile['name']} uses the category {category['name']} "
+                        f"The application profile {application_profile.name} uses the category {category.name} "
                         "undefined in 'application_classification.categories'."
                     )
                     raise AristaAvdInvalidInputsError(msg)
-                append_dict_to_list_of_dicts(new_dict=self.inputs.application_classification.categories[category["name"]]._as_dict(), list_of_dicts=categories)
+
+                if category.name:
+                    category_item = self.inputs.application_classification.categories[category.name]
+                    self.structured_config.application_traffic_recognition.categories.append(category_item)
             # Applications in application profiles
-            for application in get(application_profile, "applications", default=[]):
-                if application["name"] in self.inputs.application_classification.applications.ipv4_applications:
-                    append_dict_to_list_of_dicts(
-                        new_dict=self.inputs.application_classification.applications.ipv4_applications[application["name"]]._as_dict(),
-                        list_of_dicts=applications,
-                    )
+            for application in application_profile.applications:
+                if application.name and application.name in self.inputs.application_classification.applications.ipv4_applications:
```

Comment:
same for applicaiton name
```suggestion
                if application.name in self.inputs.application_classification.applications.ipv4_applications:
```

Reply:
shows below error:
```
Argument of type "str | None" cannot be assigned to parameter "key" of type "str" in function "__getitem__"
  Type "str | None" is not assignable to type "str"
    "None" is not assignable to "str
```

Reply:
ok I feel like we may need to add it as required. as it will fail otherwise.

Reply:
can you please update the schema?

Reply:
Done.